File: backjoon/1002.py
import math

for _ in range(int(input())):
    x1, y1, r1, x2, y2, r2 = map(int, input().split())
    distance = math.sqrt((x1-x2)**2 + (y1-y2)**2)
    large_r = max(r1, r2)
    small_r = min(r1, r2)
    # 두 원이 겹치는 경우
    if distance == 0 and r1 == r2:
        print(-1)
    # 내접하는 경우
    elif distance == r1 + r2 or distance == large_r - small_r:
        print(1)
    # 교점이 2개인 경우
    elif large_r - small_r < distance and distance < r1 + r2:
        print(2)
    # 만나지 않는 경우
    else:
        print(0)

File: baekjoon/silver/1002/1002.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
)

var (
	r *bufio.Reader = bufio.NewReader(os.Stdin)
	w *bufio.Writer = bufio.NewWriter(os.Stdout)
)

func main() {
	var t int
	var x1, y1, r1, x2, y2, r2 float64
	var dist, l, s float64

	defer w.Flush()

	fmt.Fscanf(r, "%d\n", &t)

	for i := 0; i < t; i++ {
		fmt.Fscanf(r, "%f %f %f %f %f %f\n", &x1, &y1, &r1, &x2, &y2, &r2)
		dist = math.Sqrt(math.Pow(x1-x2, 2) + math.Pow(y1-y2, 2))
		l = max(r1, r2)
		s = min(r1, r2)
		if dist == 0 && r1 == r2 {
			fmt.Fprintln(w, -1)
		} else if dist == r1+r2 || dist == l-s {
			fmt.Fprintln(w, 1)
		} else if l-s < dist && dist < r1+r2 {
			fmt.Fprintln(w, 2)
		} else {
			fmt.Fprintln(w, 0)
		}
	}
}

func min(a, b float64) float64 {
	if a < b {
		return a
	}
	return b
}

func max(a, b float64) float64 {
	if a > b {
		return a
	}
	return b
}
